### PR TITLE
Feat/update bitcoinjs lib

### PR DESCRIPTION
**Pull request by RealGoodProgrammer**
- [x] `getAccount()` now uses `ecpair` and `tiny-secp256k1` deps for generating bitcoin address
- [x] Replaced `TrasnsactionBuilder` with `Psbt`
- [x] Added `vite-plugin-wasm` and `vite-plugin-top-level-await` (required by `tiny-secp256k1`)

Since `Psbt` requires the full transaction in HEX, additional [requests](https://github.com/Adamant-im/adamant-im/pull/544/files#diff-2c46027ce891d81d49fc93869edfd6aa5df2969c156f7f7c333dd24dbac314b3R90-R93) to `GET /tx/${id}/hex` are made.

### CODE REVIEW

**Comment by bludnic** [comment]
[Update bitcoinjs-lib](https://trello.com/c/IO8SGz5y/322-update-bitcoinjs-lib)

File: src/lib/bitcoin/btc-base-api.js [hunk 1]
```diff
@@ -78,24 +83,31 @@ export default class BtcBaseApi {
...
-  createTransaction (address = '', amount = 0, fee) {
-    return this.getUnspents().then(unspents => {
-      const hex = this._buildTransaction(address, amount, unspents, fee)
+  async createTransaction(address = '', amount = 0, fee) {
+    const unspents = await this.getUnspents()
 
-      let txid = bitcoin.crypto.sha256(Buffer.from(hex, 'hex'))
-      txid = bitcoin.crypto.sha256(Buffer.from(txid))
-      txid = txid.toString('hex').match(/.{2}/g).reverse().join('')
+    // // populate with txHex <--
```
Comment (bludnic) [inline src/lib/bitcoin/btc-base-api.js:89]:
```suggestion
    // populate unspents with full transaction in HEX
```

File: src/lib/bitcoin/btc-base-api.js [hunk 2]
```diff
@@ -78,24 +83,31 @@ export default class BtcBaseApi {
...
-      return { hex, txid }
-    })
+    const hex = this._buildTransaction(address, amount, unspents, fee)
+
+    let txid = bitcoin.crypto.sha256(Buffer.from(hex, 'hex'))
+    txid = bitcoin.crypto.sha256(Buffer.from(txid))
+    txid = txid.toString('hex').match(/.{2}/g).reverse().join('')
+
+    return { hex, txid }
+    // })
```
Comment (bludnic) [inline src/lib/bitcoin/btc-base-api.js:102]: comment can be removed

File: package-lock.json [hunk 3]
```diff
@@ -13930,7 +14082,7 @@
       "version": "2.20.3",
       "resolved": "https://registry.npmjs.org/commander/-/commander-2.20.3.tgz",
       "integrity": "sha512-GpVkmM8vF2vQUkj2LvZmD35JxeJOLCwJ9cUkugyk2nuhbv3+mJvpLYYt+0+USMxE+oj+ey/lJEnhZw75x/OMcQ==",
-      "dev": true
```
Comment (bludnic) [inline package-lock.json:13933]: Still a lot of changes in `package-lock`. May need to revert the file and install again with node v18.

**Review by bludnic — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -75,8 +76,11 @@
     "stream-http": "^3.2.0",
     "stylefire": "^7.0.3",
     "throttle-promise": "^1.0.4",
+    "tiny-secp256k1": "^2.2.3",
     "tweetnacl": "^1.0.3",
     "visibilityjs": "^2.0.2",
+    "vite-plugin-top-level-await": "^1.3.1",
+    "vite-plugin-wasm": "^3.2.2",
```
Comment (bludnic) [inline package.json:83]: move to `devDependencies`

**Review by bludnic — CHANGES_REQUESTED** [review]

**Review by bludnic — APPROVED** [review]

**Review by adamant-al — COMMENTED** [review]
- Try to send txs (including taproot) through `bitcoind`, not an indexer. See 'Expose bitcoind API' https://trello.com/c/Pa45S7nJ (currently btcnode2 only).
Probably, we can switch to `bitcoind` for most requests (except tx history by address). Is it possible to optimize tx creation and don't retrieve hexes for all of unspents with `bitcoind`?

- What's the reason to add two more libs? `ecpair`, `tiny-secp256k1` (`plus vite-plugin-wasm` and `vite-plugin-top-level-await`). Why can't we use `bitcoin.ECPair.fromPrivateKey()`? Doesn't 'bitcoinjs-lib' include `ECPair` or alternative anymore?

**Comment by adamant-al** [comment]
`export default class DashApi extends BtcBaseApi`
`export default class DogeApi extends BtcBaseApi`

Have you tested DOGE and DASH?

File: src/lib/bitcoin/btc-base-api.js [hunk 5]
```diff
@@ -136,63 +147,82 @@ export default class BtcBaseApi {
...
 
-    unspents.forEach(tx => {
+    unspents.forEach((tx) => {
       const amt = Math.floor(tx.amount)
       if (transferAmount < target) {
-        txb.addInput(tx.txid, tx.vout)
+        txb.addInput({
+          hash: tx.txid,
+          index: tx.vout,
+          nonWitnessUtxo: Buffer.from(tx.txHex, 'hex')
```
[the remark this answers is not in the document]
Comment (RealGoodProgrammer) [reply src/lib/bitcoin/btc-base-api.js:169]: Yes, it always will be nonWitnessUtxo until we start to supporting BIP84

**Comment by bludnic** [comment]
> What's the reason to add two more libs? ecpair, tiny-secp256k1 (plus vite-plugin-wasm and vite-plugin-top-level-await). Why can't we use bitcoin.ECPair.fromPrivateKey()? Doesn't 'bitcoinjs-lib' include ECPair or alternative anymore?

The `bitcoinjs-lib` doesn't include ECPair in the new version, so we had to include `ecpair` and `tiny-secp256k1` dependencies (this is what `bitcoinjs-lib` suggested).

Since `tiny-secp256k1` uses WASM, additional Vite plugins are required: `vite-plugin-wasm` and `vite-plugin-top-level-await`). These are devDependencies, so it doesn't affect the bundle size.

**Comment by bludnic** [comment]
> Is it possible to optimize tx creation and don't retrieve hexes for all of unspents with bitcoind?

We will try.

**Pull request merged**